docs/api-reference/next/router.md

```diff
@@ -52,6 +52,7 @@ The following is the definition of the `router` object returned by both [`useRou
 - `domainLocales`: `Array<{domain, defaultLocale, locales}>` - Any configured domain locales.
 - `isReady`: `boolean` - Whether the router fields are updated client-side and ready for use. Should only be used inside of `useEffect` methods and not for conditionally rendering on the server. See related docs for use case with [automatically statically optimized pages](/docs/advanced-features/automatic-static-optimization.md)
 - `isPreview`: `boolean` - Whether the application is currently in [preview mode](/docs/advanced-features/preview-mode.md).
+- `key`: `string` - String uniquely representing the current history entry. Reloading will reset it.
```

Comment:
I don't think we should expose this value publicly as it could cause users to rely on it in unintended ways.

Reply:
It is suggested to publish the history key in #34980 .
I also need this feature because I want to associate state and history and store them in storage.

Reply:
The mentioned use case in the discussion wouldn't be solved here since the key is random and wouldn't give a signal to how many navigations or where in the navigation history the user is.

Reply:
What about in terms of providing functionality equivalent to remix and WICG's navigation-api? This feature would solve the case where you want to link state and history, etc.